src/Workspaces/Core/Portable/Diagnostics/DiagnosticAnalyzerExtensions.cs

```diff
@@ -48,10 +48,4 @@ public static bool SupportsSpanBasedSemanticDiagnosticAnalysis(this DiagnosticAn
         var category = analyzer.GetDiagnosticAnalyzerCategory();
         return (category & DiagnosticAnalyzerCategory.SemanticSpanAnalysis) != 0;
     }
-
-    public static bool SupportsProjectDiagnosticAnalysis(this DiagnosticAnalyzer analyzer)
-    {
-        var category = analyzer.GetDiagnosticAnalyzerCategory();
-        return (category & DiagnosticAnalyzerCategory.ProjectAnalysis) != 0;
-    }
```

Comment:
this method was never called.  so nothing ever actually checked if hte .ProjectAnalysis flag was set for an analyzer.